### PR TITLE
feat: Added Notion Source Plugin

### LINKED ISSUE #11943: feat: Add a Notion source plugin
### Describe the plugin

Notion source plugin to fetch pages and content from Notion.

### Link to API documentation

https://developers.notion.com/reference/intro

### Pull request (optional)

- [ ] I can submit a pull request

### PR DESCRIPTION
**Pull request by gagan-gaurav**
#### Summary

Created Notion Source Plugin. This PR Closes #11943

### CODE REVIEW

File: plugins/source/notion/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ - [CloudQuery Quickstart Guide](https://www.cloudquery.io/docs/quickstart)
+ - [Supported Tables](docs/tables/README.md)
+
+## Authentication
+
+In Order for CloudQuery to sync resources from your Notion setup, you will need to create a notion integration key and export the Token in NOTION_SECRET_KEY environment variable.
+How to create the notion integration key? [see here](https://developers.notion.com/docs/create-a-notion-integration#create-your-integration-in-notion). Make sure to give proper **Content Capabilities** and **User Capabilities** from Capablilities settings. Also give your integration page permissions [see here](https://developers.notion.com/docs/create-a-notion-integration#give-your-integration-page-permissions). Only pages and databases with permission will able to sync. 
+
+```bash
+export NOTION_SERCRET_KEY=<your_notion_integration_key>
```
Comment (disq) [inline plugins/source/notion/README.md:16]:
```suggestion
export NOTION_SECRET_KEY=<your_notion_integration_key>
```

File: plugins/source/notion/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+```yaml
+kind: source
+spec:
+  name: "notion"
+  path: "cloudquery/notion"
+  version: "${VERSION}"
+  destinations:
+    - "postgresql"
+  spec:
+    bearer_token: "${NOTION_SERCRET_KEY}"
```
Comment (disq) [inline plugins/source/notion/README.md:32]:
```suggestion
    bearer_token: "${NOTION_SECRET_KEY}"
```

File: plugins/source/notion/docs/tables/notion_databases_table.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+The primary key for this table is **_cq_id**.
+
+## Columns
+
+| Name          | Type          |
+| ------------- | ------------- |
+|_cq_id (PK)|`uuid`|
+|_cq_parent_id|`uuid`|
+|object|`utf8`|
+|id|`utf8`|
```
Comment (candiduslynx) [inline plugins/source/notion/docs/tables/notion_databases_table.md:14]: shouldn't ID be the PK here?

File: plugins/source/notion/docs/tables/notion_pages_table.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+The primary key for this table is **_cq_id**.
+
+## Columns
+
+| Name          | Type          |
+| ------------- | ------------- |
+|_cq_id (PK)|`uuid`|
+|_cq_parent_id|`uuid`|
+|object|`utf8`|
+|id|`utf8`|
```
Comment (candiduslynx) [inline plugins/source/notion/docs/tables/notion_pages_table.md:14]: shouldn't ID be the PK here?

File: plugins/source/notion/docs/tables/notion_users_table.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+The primary key for this table is **_cq_id**.
+
+## Columns
+
+| Name          | Type          |
+| ------------- | ------------- |
+|_cq_id (PK)|`uuid`|
+|_cq_parent_id|`uuid`|
+|id|`utf8`|
```
Comment (candiduslynx) [inline plugins/source/notion/docs/tables/notion_users_table.md:13]: shouldn't ID be the PK here?

File: plugins/source/notion/.gitignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+*.dylib
+
+# Test binary, built with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
```
Comment (candiduslynx) [inline plugins/source/notion/.gitignore:15]:
```suggestion
vendor/
```

File: plugins/source/notion/client/notion_client.go [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+func (c *NotionClient) GetUsers(nextCursor string, hasMore bool) (*users.Users, error) {
+	// Get request takes start_cursor in query string
+	queryParameter := ""
+	if hasMore {
+		queryParameter = "?start_cursor=" + nextCursor
+	}
+
+	// Create an HTTP request to path /users
+	req, err := http.NewRequest("GET", c.BaseURL+"/users"+queryParameter, nil)
```
Comment (candiduslynx) [inline plugins/source/notion/client/notion_client.go:45]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, c.BaseURL+"/users"+queryParameter, nil)
```

File: plugins/source/notion/client/notion_client.go [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+		},
+		"sort": {
+			"direction": "ascending",
+			"timestamp": "last_edited_time"
+		},
+		"page_size": 100%s
+	}`, queryParameter))
+
+	// Create an HTTP request to path /search
+	req, err := http.NewRequest("POST", c.BaseURL+"/search", bytes.NewBuffer(reqBody))
```
Comment (candiduslynx) [inline plugins/source/notion/client/notion_client.go:94]:
```suggestion
	req, err := http.NewRequest(http.MethodPost, c.BaseURL+"/search", bytes.NewBuffer(reqBody))
```

File: plugins/source/notion/client/notion_client.go [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
```
Comment (candiduslynx) [inline plugins/source/notion/client/notion_client.go:145]:
```suggestion
	req, err := http.NewRequest(http.MethodPost, c.BaseURL+"/search", bytes.NewBuffer(reqBody))
```

File: plugins/source/notion/resources/services/table.go [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/cloudquery/cloudquery/plugins/source/notion/internal/users"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+)
+
+func NotionUsersTable() *schema.Table {
+	return &schema.Table{
+		Name:      "notion_users_table",
+		Resolver:  fetchUsersTable,
+		Transform: transformers.TransformWithStruct(&users.User{}),
```
Comment (candiduslynx) [inline plugins/source/notion/resources/services/table.go:18]: To add `id` as PK:
```suggestion
		Transform: transformers.TransformWithStruct(&users.User{}, transformers.WithPrimaryKeys("Id")),
```

File: plugins/source/notion/resources/services/table.go [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+	}
+}
+
+func NotionPagesTable() *schema.Table {
+	return &schema.Table{
+		Name:      "notion_pages_table",
+		Resolver:  fetchPagesTable,
+		Transform: transformers.TransformWithStruct(&pages.Page{}),
```
Comment (candiduslynx) [inline plugins/source/notion/resources/services/table.go:26]: To add `id` as PK:
```suggestion
		Transform: transformers.TransformWithStruct(&pages.Page{}, transformers.WithPrimaryKeys("Id")),
```

File: plugins/source/notion/resources/services/table.go [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+	}
+}
+
+func NotionDatabasesTable() *schema.Table {
+	return &schema.Table{
+		Name:      "notion_databases_table",
+		Resolver:  fetchDatabasesTable,
+		Transform: transformers.TransformWithStruct(&databases.Database{}),
```
Comment (candiduslynx) [inline plugins/source/notion/resources/services/table.go:34]: To add `id` as PK:
```suggestion
		Transform: transformers.TransformWithStruct(&databases.Database{}, transformers.WithPrimaryKeys("Id")),
```

File: plugins/source/notion/README.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
+# CloudQuery notion Source Plugin
```
Comment (erezrokah) [inline plugins/source/notion/README.md:1]: We usually put the docs under the Website, see https://github.com/cloudquery/cloudquery/tree/0c9bfbeda4aa4eb45843cac53301a50b931c1bf2/website/pages/docs/plugins/sources/alicloud for an example

File: plugins/source/notion/Makefile [hunk 14]
```diff
@@ -0,0 +1,16 @@
+.PHONY: test
+test:
+	go test -timeout 3m ./...
+
+.PHONY: lint
+lint:
+	@golangci-lint run --timeout 10m
+
+.PHONY: gen-docs
+gen-docs:
```
Comment (erezrokah) [inline plugins/source/notion/Makefile:10]: We use the CLI to generate docs now, see https://github.com/cloudquery/cloudquery/blob/0c9bfbeda4aa4eb45843cac53301a50b931c1bf2/plugins/source/alicloud/Makefile#L34

If you use that script it should also copy the tables to the Website automatically (just replace `alicloud` with `notion` in the script)

File: plugins/source/notion/docs/tables/notion_users_table.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
+# Table: notion_users_table
```
Comment (erezrokah) [inline plugins/source/notion/docs/tables/notion_users_table.md:1]: These errors will resolve once the docs are moved to the Website per my comments

**Review by erezrokah — COMMENTED** [review]
Thanks @gagan-gaurav This is great 🚀 Added a few comments on the docs

File: plugins/source/notion/docs/tables/notion_databases_table.md (hunk 3 above)
Comment (gagan-gaurav) [reply plugins/source/notion/docs/tables/notion_databases_table.md:14]: yes it should. I will be changing that in next commit.

**Review by candiduslynx — APPROVED** [review]

**Comment by gagan-gaurav** [comment]
lint-grammar is failing due to a typo, i will fix that but why is the ** source plugin notion workflow ** failing, can somebody help?

**Comment by erezrokah** [comment]
@gagan-gaurav I'll take a look at the failures in a bit

File: plugins/source/notion/.gitignore [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+vendor/
+
+
+.DS_Store
+
+cq-source-notion
```
Comment (disq) [inline plugins/source/notion/.gitignore:20]:
```suggestion
notion
```
This should fix the error

Comment (gagan-gaurav) [reply plugins/source/notion/.gitignore:20]: oh ok, thanks!

**Review by disq — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
Great contribution @gagan-gaurav, added another commit so the plugin has docs on our Website and shows up under integrations https://www.cloudquery.io/integrations

File: website/pages/buy/bitbucket-extended.mdx [hunk 17]
```diff
@@ -8,7 +8,7 @@ import Head from "next/head";
   <meta httpEquiv="refresh" content="5; url='https://buy.stripe.com/bIY9AE3zP2193a83cQ'" />
 </Head>
 
-## Purchase BitBucket (Extended License - Pre-order)
```
Comment (erezrokah) [inline website/pages/buy/bitbucket-extended.mdx:11]: Seems like we forgot to update these, unrelated to this PR

**Pull request merged**